### PR TITLE
Wrap Up Encryption Assignment

### PR DESCRIPTION
**Pull request by ShawnTe**
Ready for your review! @ThatDudeHavok

### CODE REVIEW

**Comment by ShawnTe** [comment]
Saturday morning, I added revised version of file, ref last commit message. Let me know if you see anything else, @ThatDudeHavok

File: ruby/secret_agents.rb [hunk 1]
```diff
@@ -35,17 +38,54 @@ def encrypt(input_str)
 # Output = string with each character moved back one alphabetically
 #  - unless "a"
 def decrypt(input_str)
+    puts input_str
```
Comment (ThatDudeHavok) [inline ruby/secret_agents.rb]: We shouldn't have this side effect in the final code. Think we should delete it.

File: ruby/secret_agents.rb [hunk 2]
```diff
@@ -35,17 +38,54 @@ def encrypt(input_str)
...
+# we added the conditional statement.
+
+check_name = true
+while check_name
+    # gets stuff here
+    puts "Would like to decrypt or encrypt a password?"
+    method_choice = gets.chomp
+
+    if method_choice.downcase == "encrypt" || method_choice.downcase == "decrypt"
+        check_name = false
```
Comment (ThatDudeHavok) [inline ruby/secret_agents.rb]: Once they put in the correct string, the correct method is not called. Are you sure you re-push the updated code after committing our changes?

File: ruby/secret_agents.rb [hunk 3]
```diff
@@ -35,17 +38,54 @@ def encrypt(input_str)
...
+    else
+      puts "Hmmm, not understanding your answer... \"decrypt\" or \"encrypt\"?"
+    end
+end 
+
+puts "What is the password?"
+word_to_change = gets.chomp
+
+
+
```
Comment (ThatDudeHavok) [inline ruby/secret_agents.rb]: Didn't we add an if statement here? 
````ruby
if method_choice == "encrypt"
  encrypt(word_to_change)
else
  decrypt(word_to_change)
end
````

**Comment by ThatDudeHavok** [comment]
@ShawnTe: Hey, I was looking over the pull request and I noticed the request does not have the most recent code that we did. Currently, the program will never run the requested method. We have the setup to make sure that the user wants us to decrypt or encrypt a password and a variable to store it in. But we never call a method in an if statement. I made a comment in the code to be more specific. If you choose the files changed tab you'll see my comments. Also, when we worked on the code did you commit the work then push it up to GitHub?

File: ruby/secret_agents.rb (hunk 3 above)
Comment (ShawnTe) [inline ruby/secret_agents.rb]: Ah, I think I know what happened to this. Total user error. Can we leave it at that? I've added it back in.

File: ruby/secret_agents.rb [hunk 4]
```diff
@@ -8,8 +8,10 @@
 
 
  def encrypt(input_str)
+   puts input_str
```
Comment (ShawnTe) [inline ruby/secret_agents.rb]: I've removed this puts as well.

**Comment by ShawnTe** [comment]
Process Q, (in reference to your Q about pushing commit to GitHub). In our workflow, I was _not_ pushing to GitHub each time I did a commit. Should I have been?

**Comment by ShawnTe** [comment]
Answering my own Q re: pushing after commits. I hadn't been b/c I'd seen before that all the commits get lumped into one pull request anyways. AND... there's no online backup, so to speak, of each of those commits along the way for collaborators or my future self to refer to,  so YES, I'd think they should be pushed each commit.

**Comment by ShawnTe** [comment]
Comments and new commit added to pull request, @ThatDudeHavok

**Comment by ThatDudeHavok** [comment]
I was talking about pushing specifically after you had finished making your final edits. Pushing up after ever single commit will make the coding process way too long. As a general rule, if you want every commit in a branch, you should always push up your current branch before making a pull request for the branch in question. I was asking because yesterday I didn't see the if statement to actually call the correct function. But now everything looks good, great job and it was awesome to work with you.

**Pull request merged**